auction-server/src/auction.rs

```diff
@@ -157,26 +161,31 @@ impl Transformer for LegacyTxTransformer {
 pub async fn submit_bids(
     express_relay_contract: Arc<SignableExpressRelayContract>,
     permission: Bytes,
-    multicall_data: Vec<MulticallData>,
+    // multicall_data: Vec<MulticallData>,
+    bids: Vec<SimulatedBid>,
 ) -> Result<H256, ContractError<SignableProvider>> {
-    let call = express_relay_contract.multicall(permission, multicall_data);
-    let mut gas_estimate = call.estimate_gas().await?;
-
-    let gas_multiplier = U256::from(2); //TODO: smarter gas estimation
-    gas_estimate *= gas_multiplier;
-    let call_with_gas = call.gas(gas_estimate);
-    let send_call = call_with_gas.send().await?;
-
-    Ok(send_call.tx_hash())
+    let gas_estimate = bids.iter().fold(U256::zero(), |sum, b| sum + b.gas_limit);
+    let tx_hash = express_relay_contract
+        .multicall(
+            permission,
+            bids.into_iter().map(|b| (b, false).into()).collect(),
+        )
+        .gas(gas_estimate + EXTRA_GAS_FOR_SUBMISSION)
```

Comment:
I would rather multiply this by 1.2 or some number again instead of having a constant offset